### ISSUE TITLE
Inlay Hint Support

### ISSUE THREAD

**Issue by Popkornium18** [bash-lsp/bash-language-server#1173]
### What is the problem this feature will solve?

Bash function calls with lots of arguments are hard to parse

```bash
somefunction()
{
  local -r foo=$1
  local -r bar=$1
  local -r foo2=$1
  local -r bar2=$1
  # Do something useful
}

somefunction one two three four
```

### What is the feature you are proposing to solve the problem?

Inlay Hints try to solve this problem by displaying a "virtual text" with the argument name. Here is a lua example:

![grafik](https://github.com/bash-lsp/bash-language-server/assets/46488511/a17061f9-4586-40f4-85a7-0f562e32469f)

With inlay hints it could be displayed in the IDE like this:

```
somefunction foo: one bar: two foo2: three bar2: four
```

This is obviously a little tougher with bash, as there is no strict function argument definition that could be parsed. Possible solutions would be parsing (and possibly establishing) a docstring format that explains the arguments, or reacting to the pattern where the first lines of the function are renaming `$1` and so on.

**Comment by Bdrake317** [comment]
Open

**Comment by Bdrake317** [comment]
> I would rather rely on the renaming `$1, $2, ...` parameters because in such a way:
> 
> 
> 
> - it's possible to use these renamed parameters because they are variables actually
> 
> - it allows writing more readable code which becomes self-documented
> 
> 
> 
> 😄 
> 
> 

@

**Comment by Bdrake317** [comment]
> > I would rather rely on the renaming `$1, $2, ...` parameters because in such a way:
> > 
> > 
> > 
> > - it's possible to use these renamed parameters because they are variables actually
> > 
> > - it allows writing more readable code which becomes self-documented
> > 
> > 
> > 
> > 😄 
> > 
> > 
> 
> @Bdrake317

**Comment by Popkornium18** [comment]
Since the virtual text can be anything you want, I think the most readable variant would be:

```
somefunction foo=one bar=two foo2=three bar2=four
```
